packages/scenes/src/components/VizPanel/VizPanelMenu.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { PanelMenuItem } from '@grafana/data';
+import { Menu } from '@grafana/ui';
+import { SceneObjectBase } from '../../core/SceneObjectBase';
+import { SceneComponentProps, SceneObjectStatePlain } from '../../core/types';
+
+interface VizPanelMenuState extends SceneObjectStatePlain {
+  items: PanelMenuItem[];
+}
+
+export class VizPanelMenu extends SceneObjectBase<VizPanelMenuState> {
+  static Component = VizPanelMenuRenderer;
+
+  public constructor(state: Partial<VizPanelMenuState>) {
+    super({
+      ...state,
+      items: state.items || [],
```

Comment:
```suggestion
      items: state.items ?? [],
```

nitpick: since items is required (not optional), this code / default setting seems unnecessary